src/unit/test_networking.c

```diff
@@ -4,6 +4,305 @@
+        block1->size = 128;
+        block1->used = 64;
+        block2->size = 128;
+        block2->used = 32;
+        memset(block1->buf, 'A', 64);
+        memset(block2->buf, 'B', 32);
+
+        /* Setup client state */
+        listAddNodeTail(server.repl_buffer_blocks, block1);
+        listAddNodeTail(server.repl_buffer_blocks, block2);
+        c->ref_repl_buf_node = listFirst(server.repl_buffer_blocks);
+        c->ref_block_pos = 0;
+        c->bufpos = 0;
+
+        writeToReplica(c);
+
+        TEST_ASSERT(c->nwritten == 96); /* 64 + 32 */
+        TEST_ASSERT(mock_conn->written == 96);
+        TEST_ASSERT(memcmp(mock_conn->buffer, block1->buf, 64) == 0);
+        TEST_ASSERT(memcmp(mock_conn->buffer + 64, block2->buf, 32) == 0);
+        TEST_ASSERT((c->write_flags & WRITE_FLAGS_WRITE_ERROR) == 0);
+
+        /* Cleanup */
+        zfree(mock_conn->buffer);
+        zfree(mock_conn);
+        zfree(block1);
+        zfree(block2);
+        listEmpty(server.repl_buffer_blocks);
+    }
+
+    /* Test 3: Write error */
+    {
+        mockConnection *mock_conn = connCreateMock();
+        mock_conn->error = 1; /* Simulate write error */
+        mock_conn->buffer = zmalloc(1024);
+        mock_conn->buf_size = 1024;
+        mock_conn->written = 0;
+        c->conn = (connection *)mock_conn;
+
+        /* Create replication buffer block */
+        replBufBlock *block = zmalloc(sizeof(replBufBlock) + 128);
+        block->size = 128;
+        block->used = 64;
+        memset(block->buf, 'A', 64);
+
+        /* Setup client state */
+        listAddNodeTail(server.repl_buffer_blocks, block);
+        c->ref_repl_buf_node = listFirst(server.repl_buffer_blocks);
+        c->ref_block_pos = 0;
+        c->bufpos = 0;
+
+        writeToReplica(c);
+
+        TEST_ASSERT(c->nwritten <= 0);
+        TEST_ASSERT((c->write_flags & WRITE_FLAGS_WRITE_ERROR) != 0);
+
+        /* Cleanup */
+        listEmpty(server.repl_buffer_blocks);
+        zfree(mock_conn->buffer);
+        zfree(mock_conn);
+        zfree(block);
+    }
+
+    /* Cleanup */
+    listRelease(server.repl_buffer_blocks);
+    listRelease(c->reply);
+    zfree(c);
+
+    return 0;
+}
+
+int test_postWriteToReplica(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    client *c = zcalloc(sizeof(client));
+    server.repl_buffer_blocks = listCreate();
+    c->reply = listCreate();
+    createReplicationBacklog();
+
+    /* Test 1: No write case */
+    {
+        c->nwritten = 0;
+        server.stat_net_repl_output_bytes = 0;
+
+        postWriteToReplica(c);
+
+        TEST_ASSERT(server.stat_net_repl_output_bytes == 0);
+    }
+
+    /* Test 2: Single block partial write */
+    {
+        replBufBlock *block = zmalloc(sizeof(replBufBlock) + 128);
+        block->size = 128;
+        block->used = 100;
+        block->refcount = 1;
+
+        listAddNodeTail(server.repl_buffer_blocks, block);
```

Comment:
Looks like this code is duplicated, I suggesting refactoring this into:
`
appendReplBufBlock(size_t size, size_t used) ;
`

Reply:
The size and used values differ each time